btx/processing/rawimagetimetool.py

```diff
@@ -0,0 +1,575 @@
+                                        fwhm: np.array = None, order: int = 2):
+        """! Fit a polynomial calibration curve to a list of edge pixel positions
+            vs delay. In the future this will implement edge acceptance/rejection to
+            improve the calibration. Currently only fits within a certain x pixel
+            range of the camera.
+
+        @param delays (list) List of TT target times (in ns) used for calibration.
+        @param edges (list) List of corresponding detected edge positions on camera.
+        @param ampls (list) Convolution amplitudes used for edge rejection.
+        @param fwhms (list) Full-width half-max of convolution used for edge rejection.
+        @param order (int) Order of polynomial to fit.
+        """
+        # Should only fit a certain X range of the data.
+        # Ad hoc something like 250 - 870 may be appropriate.
+        # This will almost certainly depend on experimental conditions, alignment
+        # target choice, etc.
+
+        # Restructure this gross conditional block
+        inrange = False
+        # Delays and edges should be the same length if error checking in the
+        # process_calib_run function works properly
+        #@todo implement amplitude- and fwhm-based selection and compare to this
+        # simplified version
+        self.edges = edges
+        self.delays = delays
+        if (edges > 250).any():
+            delays_fit = delays[edges > 250]
+            edges_fit = edges[edges > 250]
+            if (edges_fit < 870).any():
+                delays_fit = delays_fit[edges_fit < 870]
+                edges_fit = edges_fit[edges_fit < 870]
+                inrange = True
+
+                # For testing
+                self.edges_fit = edges_fit
+                self.delays_fit = delays_fit
+        if inrange:
+            print('Fitting calibration between 250 and 870 pixels.')
+            self._model = np.polyfit(edges_fit, delays_fit, order)
+        else:
+            print('Fit data is out of 250-870 pixel range. Results questionable.')
+            self.edges_fit = self.edges
+            self.delays_fit = self.delays
+            self._model = np.polyfit(edges, delays, order)
+
+    def ttstage_code(self, hutch: str) -> str:
+        """! Return the correct code for the time tool delay (in ns) for a given
+        experimental hutch.
+
+        @param hutch (str) Three letter hutch name. E.g. mfx, cxi, xpp
+
+        @return code (str) Epics code for accessing hutches time tool delay.
+        """
+        h = hutch.lower()
+        num = ''
+
+        if h == 'xpp':
+            num = '3'
+        elif h == 'xcs':
+            num = '4'
+        elif h == 'mfx':
+            num = '45'
+        elif h == 'cxi':
+            num = '5'
+
+        if num:
+            return f'LAS:FS{num}:VIT:FS_TGT_TIME_DIAL'
+        else:
+            raise InvalidHutchError(hutch)
```

Comment:
Great! Let's discuss again how this could be part of a bigger/centralized PV library.